ironfish-rust/src/serializing/aead.rs

```diff
@@ -58,10 +60,40 @@ pub(crate) fn decrypt<const SIZE: usize>(
     Ok(plaintext)
 }
 
+
+/// WARNING:
+/// 1. Decryption can only occur for 1-n bytes, not n-m bytes 
+/// 2. There is no MAC tag check here, so this is not secure.
+/// 3. There is no guarantee the encrypted text is valid, it will return a "value" whether the key is valid or not
+/// 
+/// this is a partial decryption, and should only be used for cases where
+/// potential tampering is not a concern. For example, trial decryption of
+/// a note for determining if that note is relevant to an account, where the full ciphertext
+/// will subsequently be downloaded 
+pub fn decrypt_partial(
+    key: &[u8; 32],
+    truncated_ciphertext: &Vec<u8>,
+) -> Vec<u8> {
+    let mut truncated_encrypted_text = truncated_ciphertext.clone();
+    truncated_encrypted_text.copy_from_slice(&truncated_ciphertext[..]);
+
+    let mut keystream = ChaCha20::new(key.as_ref().into(), [0u8; 12][..].into());
+    keystream.seek(64);
```

Comment:
This seek value needs to change based on the position of the plaintext (for example if you want to decrypt bytes 32-64 instead of 0-32, this would need to seek to 96. Hardcoded here for test